### PR TITLE
DPMMA-2372 Email usages 

### PR DESCRIPTION
**Pull request by patrykgruszka**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ N ]
| New feature/enhancement? (use the a.x branch)      | [ Y ]
| Deprecations?                          | [ N ]
| BC breaks? (use the c.x branch)        | [ N ]
| Automated tests included?              | [ Y ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 

#### Description:
This PR adds information about email usage in segments, campaigns, forms, points and reports. Similar functionality already works for segments. 

This PR also brings UI improvements for this component - the table has been replaced with buttons so it's more compact and easier to click on the link.
image

The email usages are loaded asynchronously, so this should not affect the load time of the email view. 

A message has been added if the email/segment is not used elsewhere:
- For Emails: This email is not used by other components. 
- For Segments: Looks like this segment is not in use. 

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Add a new Email
3. Use the email in segment filter / campaign action / point trigger / point action / report
4. Check that the email usage is visible in the email view

### CODE REVIEW

**Review by kuzmany — APPROVED** [review]
Nice improvements. I did basic tests with emails and segments, and it works properly 👍

**Review by andersonjeccel — APPROVED** [review]
It looks great and definitely is an improvement. 
! We need to add this to the ux/ui-redesign branch before finishing the interface redesign.

File: app/bundles/EmailBundle/Controller/AjaxController.php [hunk 1]
```diff
@@ -317,4 +318,27 @@ public function heatmapAction(Request $request, EmailModel $model): JsonResponse
...
         ]);
     }
+
+    public function getEmailUsagesAction(Request $request, EmailDependencies $emailDependencies): JsonResponse
+    {
+        $emailId = (int) $request->query->get('id');
+
+        if (0 === $emailId) {
+            return $this->sendJsonResponse([
+                'success' => 0,
```
Comment (escopecz) [inline app/bundles/EmailBundle/Controller/AjaxController.php]: I'd suggest to stop using the `success` flag in the response and use HTTP statuses instead. I know it's used elsewhere in Mautic but it's a bad habit.

Comment (patrykgruszka) [reply app/bundles/EmailBundle/Controller/AjaxController.php]: You're right, I changed it.

**Pull request merged**